server/cache.go

```diff
@@ -277,53 +282,21 @@ func (r *regionsInfo) heartbeatVersion(region *metapb.Region) (bool, *metapb.Reg
 	return true, searchRegion, nil
 }
 
-func (r *regionsInfo) heartbeatConfVer(region *metapb.Region) (*pdpb.ChangePeer, bool, error) {
+func (r *regionsInfo) heartbeatConfVer(region *metapb.Region) (bool, error) {
 	// ConfVer is handled after Version, so here
 	// we must get the region by ID.
 	cacheRegion := r.regions[region.GetId()]
 	if err := checkStaleRegion(cacheRegion, region); err != nil {
-		return nil, false, errors.Trace(err)
+		return false, errors.Trace(err)
 	}
 
 	if region.GetRegionEpoch().GetConfVer() > cacheRegion.GetRegionEpoch().GetConfVer() {
 		// ConfChanged, update
 		r.updateRegion(region)
-		return r.getChangePeer(cacheRegion, region), true, nil
```

Comment:
It's not necessary to handle it. The original code just records the `ChangePeer` event, but that's already recorded when the corresponding operator has finished, there is no reason to record this event here again.